### ISSUE TITLE
Selenium queries to UWP app intermittently hanging up winappdriver.exe

### ISSUE THREAD

**Issue by jsbakker** [Microsoft/WinAppDriver#296]
**Steps**:
- Run winappdriver in a cmd window `winappdriver 127.0.0.1 4723/wd/hub`
- Run a suite of Ruby test scripts against a UWP application using selenium-webdriver API

**Result**:
Sometimes the query never returns execution. Observations:
- There is zero activity output in the winappdriver console window
- The test script will synchronously wait until winappdriver.exe is un-stuck

**Workaround**:
Press Ctrl+C in the winappdriver.exe console window, then restart winappdriver.exe (Up Arrow, Enter). When this happens, the console output window becomes active again and tests continue to execute.

**Notes**:
This does not occur on any specific query; it can happen randomly on regularly passing tests. It pauses up in different places, whether it's a `find_element` call or a `start_driver` call.

If it freezes on a wait.until {driver.find_element blah blah} call, it is easy to spot if you know to look in the console window. Wait until usually shows a bunch of responses in the winappdriver console window until the element is found. If winappdriver is hung though, then there is no output.

This has been seen on 2 different Windows 10 machines. It was encountered multiple times in a day on both machines.

I will try to get an output log next time I see this (but currently I am not running with log write + real time stdout to see when the hang occurs).

**Comment by jsbakker** [comment]
**Workaround 2**:
If the app is running (e.g. issue encountered after start_driver), then closing the app using the 'X' button on the titlebar appears to get the winappdriver unstuck. Here is a portion of a log of getting stuck on an "Authentication" screen for several minutes and closing the app.

[winappdriver_close.txt](https://github.com/Microsoft/WinAppDriver/files/1452155/winappdriver_close.txt)

Also, on the test suite side, this is the error the tests get when trying to talk to winappdriver:

```
    Given I am logged in as automation test user after a fresh install # Superliminal/Superliminal.UWP/features/step_definitions/site_authentication_steps.rb:1
      Failed to open TCP connection to 127.0.0.1:4723 (No connection could be made because the target machine actively refused it. - connect(2) for "127.0.0.1" port 4723) (Errno::ECONNREFUSED)
```

**Comment by hassanuz** [comment]
Hi @jsbakker, 

Is it possible for you to share a test script, or any samples that we can try reproducing this issue with? 

Thanks

**Comment by jsbakker** [comment]
I was able to solve this issue. It turned out that the way we were loading the WebView.Source via programmatic redirect for our Xamarin.Forms WebView control (which uses the UWP web view underneath) was causing some freezing within the web view itself, while the rest of the application UI remained responsive. If we are using winappdriver to look for something in this frozen-up webview, then winappdriver would freeze up too.

We ended up removing the redirects which sometimes froze up the WebView control, and after that, we haven't seen winappdriver freeze up.
